server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceMetrics.java

```diff
@@ -86,6 +110,35 @@ private List<LongWithAttributes> getUnassignedShardsMetrics() {
         return getIfPublishing(unassignedShards);
     }
 
+    private List<DoubleWithAttributes> getNodeWeightStats() {
+        if (nodeIsMaster == false) {
+            return List.of();
+        }
+        var stats = weightStatsPerNodeRef.get();
+        List<DoubleWithAttributes> doubles = new ArrayList<>(stats.size());
+        for (var node : stats.keySet()) {
+            var stat = stats.get(node);
+            doubles.add(
+                new DoubleWithAttributes(
+                    stat.nodeWeight(),
+                    Map.of(
+                        "node_id",
+                        node.getId(),
+                        "node_name",
+                        node.getName(),
+                        "shard_count",
+                        stat.shardCount(),
+                        "disk_usage_in_bytes",
+                        stat.diskUsageInBytes(),
+                        "write_load",
+                        stat.writeLoad()
+                    )
```

Comment:
I'm pretty sure we shouldn't be recording continuous values as attributes, as I understand it each distinct combination of attribute values effectively creates a separate metric, so having numbers like write load, disk usage, shard count in there will cause an explosion of metrics. I believe attributes should (mostly) be limited to low-cardinality, discrete values. See conversation about that [here](https://elastic.slack.com/archives/C5G0S6QD7/p1726650385124389?thread_ts=1726648548.317119&cid=C5G0S6QD7).

I wonder if instead we should track those values as separate metrics, that would also allow them to be plotted. We could have a chart that plotted the constituent values along with the resulting weight.

Reply:
This is the only discussion I could find on it but I think it's not splunk-specific.

https://docs.splunk.com/observability/en/gdi/opentelemetry/tags.html#considerations-on-metric-cardinality

Reply:
I agree. I do not think we can even plot labels, can we?
I suspect this could be a metric per component + the combined one.

Reply:
> I'm pretty sure we shouldn't be recording continuous values as attribute values

You might be right. I quickly tried to plot some of the `es.autoscaling.indexing.node_ingest_load.current` labels (that's the example that made me think I can use labels) and I got a `This field does not work with the selected function.` when trying to use last_value function. not sure if it is because of the label types. Nonetheless, to be on the safe side, I'll break this up into difference metrics.

Reply:
> I got a This field does not work with the selected function. when trying to use last_value function.

👍 good to know.

Reply:
I've split them in e79b5e1cd7b50ec01c4dde28ca8c469c1a4cb414.